apollo-router/src/plugins/demand_control/mod.rs

```diff
@@ -187,6 +193,7 @@ impl DemandControlError {
             DemandControlError::ActualCostTooExpensive { .. } => "COST_ACTUAL_TOO_EXPENSIVE",
             DemandControlError::QueryParseFailure(_) => "COST_QUERY_PARSE_FAILURE",
             DemandControlError::ResponseTypingFailure(_) => "COST_RESPONSE_TYPING_FAILURE",
+            DemandControlError::InvalidSubgraphQuery(_) => "GRAPHQL_VALIDATION_FAILED",
```

Comment:
we need another error code than this. If the cleient query reached the planner, then it was valid, so the client should not assume its query had an issue